endpoints/authentication/src/main.cpp

```diff
@@ -163,13 +186,23 @@ namespace irods::http::handler
 		return {std::move(username), std::move(password)};
 	}
 
+	auto is_oidc_running_as_client() -> bool
+	{
+		static const auto oidc_stanza_exists = irods::http::globals::configuration().contains(
+			nlohmann::json::json_pointer{"/http_server/authentication/openid_connect"});
+
+		if (oidc_stanza_exists) {
+			static const auto is_client{irods::http::globals::oidc_configuration().at("run_as_protected_resource").get<bool>()};
+			return is_client;
+		}
+
+		return false;
+	}
+
 	IRODS_HTTP_API_ENDPOINT_ENTRY_FUNCTION_SIGNATURE(authentication)
 	{
 		if (_req.method() == boost::beast::http::verb::get) {
-			static const auto oidc_stanza_exists = irods::http::globals::configuration().contains(
-				nlohmann::json::json_pointer{"/http_server/authentication/openid_connect"});
-
-			if (!oidc_stanza_exists) {
+			if (!is_oidc_running_as_client()) {
```

Comment:
I think this doesn't do what we want it to...

Reply:
Please elaborate.

Reply:
I thought there was a logic issue in `is_oidc_running_as_client(...)`.

Reply:
And so, what's the verdict now?

Reply:
I'm going to double-check it, but I think it works. Besides, with the schema change, it needs to be updated anywho.

Reply:
Done.